cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -199,29 +200,40 @@ def configure_ua(token=None, enable=None, config=None):
                 ", ".join('"{}"'.format(param) for param, _ in enable_errors)
             )
         )
-    attach_cmd = ["ua", "attach", token]
+
+    if enable:
+        attach_cmd = ["ua", "attach", "--no-auto-enable", token]
+    else:
+        attach_cmd = ["ua", "attach", token]
     LOG.debug("Attaching to Ubuntu Advantage. %s", " ".join(attach_cmd))
     try:
-        subp.subp(attach_cmd)
+        # Allow `ua attach` to fail in already attached machines
+        subp.subp(attach_cmd, rcs={0, 2})
     except subp.ProcessExecutionError as e:
         msg = "Failure attaching Ubuntu Advantage:\n{error}".format(
             error=str(e)
         )
         util.logexc(LOG, msg)
         raise RuntimeError(msg) from e
+
     enable_errors = []
     for service in enable:
         try:
             cmd = ["ua", "enable", "--assume-yes", service]
```

Comment:
Can we adapt this cmd so we parse --format json output which should be more stable across languages than the CLI stdout.

```suggestion
            cmd = ["ua", "enable", "--assume-yes", service, "--format", "json"]
```

Reply:
Tracking in https://warthogs.atlassian.net/browse/SC-1253